Protos/V1/agent_common.proto

```diff
@@ -70,13 +79,152 @@ message Result {
+  message ResultCreate {
+    string name = 1; /** The result name. Given by the client. */
+  }
+  repeated ResultCreate results = 1; /** The list of results to create. */
+  string session_id = 2; /** The session in which create results. */
+  string communication_token = 3; /** Communication token received by the worker during task processing */
+}
+
+/**
+* Result metadata
+*/
+message ResultMetaData {
+  string session_id = 1; /** The session ID. */
+  string name = 2; /** The result name. */
+  string owner_task_id = 3; /** The owner task ID. */
+  result_status.ResultStatus status = 4; /** The result status. */
+  google.protobuf.Timestamp created_at = 5; /** The result creation date. */
+}
+
+/*
+* Response for creating results without data
+*/
+message CreateResultsMetaDataResponse {
+  repeated ResultMetaData results = 1; /** The list of metadata results that were created. */
+  string communication_token = 2; /** Communication token received by the worker during task processing */
+}
+
+/**
+* Request to create tasks.
+*/
+message SubmitTasksRequest {
+  message TaskCreation {
+    repeated string expected_output_keys = 1; /** Unique ID of the results that will be produced by the task. Results are created using ResultsService. */
+    repeated string data_dependencies = 2; /** Unique ID of the results that will be used as datadependencies. Results are created using ResultsService. */
+    string payload_id = 3; /** Unique ID of the result that will be used as payload. Results are created using ResultsService. */
+    TaskOptions task_options = 4; /** Optionnal task options. */
+  }
+
+  string session_id = 1; /** The session ID. */
+  TaskOptions task_options = 2; /** The options for the tasks. Each task will have the same. Options are merged with the one from the session. */
+  repeated TaskCreation task_creations = 3; /** Task creation requests. */
+  string communication_token = 4; /** Communication token received by the worker during task processing */
+}
+
+/**
+* Response to create tasks.
+*
+* expected_output_ids and data_dependencies must be created through ResultsService.
+*
+* Remark : this may have to be enriched to a better management of errors but
+* will the client application be able to manage a missing data dependency or expected output ?
+*/
+message SubmitTasksResponse {
+  message ResultIdName {
+    string name = 1; /** Result given name. */
+    string result_id = 2; /** Result Unique ID. */
+  }
+
+  message TaskInfo {
+    string task_id = 1; /** The task ID. */
+    repeated string expected_output_ids = 2; /** The expected output IDs. A task have expected output IDs. */
+    repeated string data_dependencies = 3; /** The data dependencies IDs (inputs). A task have data dependencies. */
+    string payload_id = 4; /** Unique ID of the result that will be used as payload. Results are created implicitly. */
+    repeated ResultIdName id_name = 5; /** List of association between given names and result IDs. */
+  }
```

Comment:
Why have multiple definitions of taskinfo in different message types ?

Reply:
mutualizing the message means introducing a breaking change in the existing message.
I'm not sure we want to introduce more coupling between messages.

Reply:
Fair enough

Reply:
@lemaitre-aneo your thougts ?

Reply:
@aneojgurhem I agree with you. We should delay this discussion until we break compatibility.